File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    
    alpha = {}
    
    for key in keymap:
        for i, a in enumerate(key):
            if a in alpha:
                alpha[a] = min(alpha[a], i + 1)
            else:
                alpha[a] = i + 1
    
    for target in targets:
        cnt = 0
        for t in target:
            if t in alpha:
                cnt += alpha[t]
            else:
                cnt = -1
                break
        
        if cnt < 0:
            answer.append(-1)
        else:
            answer.append(cnt)
    
    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    var answer = [];
    
    const dict = {};
    for (let key of keymap) {
        for (let i = 0; i < key.length; i++) {
            if (key[i] in dict) {
                if (dict[key[i]] >= i + 1) dict[key[i]] = i + 1;
            } 
            else dict[key[i]] = i + 1;
        }
    }
    
    for (let target of targets) {
        let cnt = 0;
        for (let tg of target) {
            if (tg in dict) cnt += dict[tg] ;
            else {
                cnt = -1;
                break;
            }
        }
        answer.push(cnt);
    }
    
    return answer;
}